parser/src/java/org/apache/hadoop/hive/ql/parse/IdentifiersParser.g

```diff
@@ -988,6 +988,16 @@ nonReserved
     | KW_TAG
     | KW_FAST_FORWARD
     | KW_OPTIMIZE
+    | KW_APPLICATION
+    | KW_COMPACT_ID
+    | KW_DATACONNECTOR
+    | KW_DATACONNECTORS
+    | KW_DDL
+    | KW_FORCE
+    | KW_OLDER
+    | KW_PKFK_JOIN
+    | KW_THAN
+    | KW_TIMESTAMPLOCALTZ
 ;
```

Comment:
Tickets and fix versions

- KW_APPLICATION: HIVE-18004(fix version=3.0.0)
- KW_COMPACT_ID: HIVE-26580(4.0.0)
- KW_DATACONNECTOR  & KW_DATACONNECTORS: HIVE-24396(4.0.0)
- KW_DDL: HIVE-24596(4.0.0)
- KW_FORCE: HIVE-4367(4.0.0)
- KW_OLDER & KW_THAN: HIVE-27906(4.0.0)
- KW_PKFK_JOIN: HIVE-23817(maybe 4.0.0)
- KW_TIMESTAMPLOCALTZ: HIVE-16614(3.0.0)
    - [This could have replaced KW_TIMESTAMPTZ](https://github.com/apache/hive/commit/34eebff194e81180202d198200e84058c4910d95#diff-0873af31a12d73e96355bffe6b8a1435c958c744231f0ccd3f80d057920cb8ce)